src/index.ts

```
@@ -1,14 +1,35 @@
 //Define class here
-console.log("hello there!");
+const mockUsers = [
+  { id: 0, name: "minoka", age: 31, favoriteColor: "green" },
+  { id: 1, name: "ted", age: 23, favoriteColor: "black" },
+  { id: 2, name: "ron", age: 41, favoriteColor: "blue" },
+  { id: 3, name: "tim", age: 27, favoriteColor: "red" },
+];
 
 class UserAPI {
-  constructor() {}
+  constructor(users) {
+    this.list = users || [];
+  }
+  //random id generator
+  randomId = () => {
+    return Math.random().toString(24).slice(2);
+  };
   // these are the methods needed for basic CRUD
-  addUser(name) {
-    console.log(`inside addUser ${name}`);
+  addUser(user) {
+    const newUser = { id: this.randomId(), ...user };
+    this.list.push(newUser);
+    return newUser;
   }
   getUserById(id) {
-    console.log(`looking for user with Id of ${id}`);
+    const targetUser = this.list.find((user) => user.id === id);
+    if (!targetUser) {
+      return new Error("There are no users found with that id.");
```

read up on throwing exceptions in javascript rather than returning an object. this function should return a user or throw an exception

---

as far as throwing an error goes, I cant think of a way to console.log() the thrown error. but when i returned the err in each try catch the correct message was displaying.